theano/tensor/nnet/nnet.py

```diff
@@ -401,6 +396,11 @@ def make_node(self, dy, sm, axis=-1):
             dy = tensor.shape_padleft(dy, n_ones=1)
         if sm.ndim == 1:
             sm = tensor.shape_padleft(sm, n_ones=1)
+        # Check if axis is a scalar
+        try:
+            opt.get_scalar_constant_value(axis)
+        except tensor.NotScalarConstantError:
+            raise ValueError('axis must be a scalar. Use abstract_softmax instead Got ', axis.type)
         return Apply(self, [dy, sm, axis], [sm.type()])
```

Comment:
After discussion, we agreed to use a params type: http://deeplearning.net/software/theano_versions/dev/library/gof/params_type.html instead of the symbolic axis.